tests/unit/test_protocol_errors.py

```diff
@@ -0,0 +1,105 @@
+"""
```

Comment:
Yeah, the error codes are completely up to the implementation to use as they see fit. In general, the specification of error handling in the protocol is pretty much non-existent (which is definitely not a good idea!).